include/libtorrent/aux_/typed_span.hpp

```diff
@@ -64,56 +64,56 @@ namespace libtorrent { namespace aux {
 
 		template <typename U = underlying_index, typename Cond
 			= typename std::enable_if<std::is_signed<U>::value>::type>
-		typed_span<T> first(underlying_index n) const
+		typed_span<T, IndexType> first(underlying_index n) const
```

Comment:
actually, you should be able to just say ``typed_span``. iirc, the default is to imply the same template arguments as the class' own arguments. would you mind trying that?